### PR TITLE
[Quant] X86 qengine always uses fbgemm kernels on OS other than Linux

### PR DESCRIPTION
**Pull request by Xia-Weiwen**
**Summary**
X86 quantization backend (qengine) with oneDNN kernels has not been validated on OS other than Linux. So, let it fall back to fbgemm if OS is not Linux. This makes sure the behavior is the same on Windows/Mac as the previous default fbgemm qengine on x86 CPUs.

**Test plan**
CI checks.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

**Comment by Xia-Weiwen** [comment]
Hi @jerryzh168. Could you please review this PR? Thanks.
For x86 backend, we plan to use onednn kernels only on Linux as we have not validated the performance yet on other OSs.

**Review by jerryzh168 — APPROVED** [review]
thanks, please feel free to land the onednn only changes directly. or is this blocked by a review from pytorch dev?

**Comment by Xia-Weiwen** [comment]
> thanks, please feel free to land the onednn only changes directly. or is this blocked by a review from pytorch dev?

Yes. I think such PRs need review from Meta side. For example, https://github.com/pytorch/pytorch/pull/91934 cannot be merged with approval from our team only (merged failed). And this PR is related to x86 backend, not actually onednn only I think, so we think it also needs approval from Meta side.

**Pull request closed without merging**
